util/gitutil/git_cli.go

```diff
@@ -189,6 +189,12 @@ func (cli *GitCLI) Run(ctx context.Context, args ...string) (_ []byte, err error
 			"GIT_CONFIG_NOSYSTEM=1", // Disable reading from system gitconfig.
 			"HOME=/dev/null",        // Disable reading from user gitconfig.
 			"LC_ALL=C",              // Ensure consistent output.
+			"HTTP_PROXY=" + os.Getenv("HTTP_PROXY"),
```

Comment:
Afaics this should define new variables with empty string values if the env value is not defined. `os.LookupEnv` handles the case of detecting if env is defined or not.

Reply:
True, I've updated the MR to use `os.LookupEnv`

Reply:
FYI: i've put `proxyEnvVars` out of the loop to not instantiate and free it each loop iteration.

That said I think that it would be better to define the whole `env` slice out of the loop and then assign it to `cmd.Env` during loop iteration. This way we don't call `os.Getenv`, `os.LookupEnv` nor `getGitSSHCommand` every single loop iteration.